src/app/pages/things/channels/details/channels.details.component.ts

```diff
@@ -109,12 +110,14 @@ export class ChannelsDetailsComponent implements OnInit {
   }
 
   getchannelMessages() {
-    this.messagesService.getMessages(this.channel.id, this.connections[0].key).subscribe(
-      (respMsg: any) => {
-        if (respMsg.messages) {
-          this.messages = respMsg.messages;
-        }
-      },
-    );
+    if (this.connections.length > 0) {
+      this.messagesService.getMessages(this.channel.id, this.connections[0].key).subscribe(
+        (respMsg: any) => {
+          if (respMsg.messages) {
+            this.messages = respMsg.messages;
```

Comment:
Maybe it's better like this: 

`this.messages = respMsg.messages || this.messages;`